### PR TITLE
Add Gas Estimation Guide

### LINKED ISSUE #7299: Create a new Guide about Gas Estimation

### PR DESCRIPTION
**Pull request by danforbes**
Closes #7299

### CODE REVIEW

File: docs/docs/guides/04_transactions/gas-and-fees.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+Web3.js transactions may specifying a gas limit (the maximum amount of gas they are able to consume) by providing the [`Transaction.gas` property](/api/web3/namespace/types#gas). If the specified gas limit is less than the actual amount of gas required to execute the transaction, the transaction will consume an amount of gas equal to the gas limit, which is not refunded, before failing and reverting any state changes made by the transaction.
```
Comment (krzysu) [inline docs/docs/guides/04_transactions/gas-and-fees.md]:
```suggestion
Web3.js transactions may specify a gas limit (the maximum amount of gas they are able to consume) by providing the [`Transaction.gas` property](/api/web3/namespace/types#gas). If the specified gas limit is less than the actual amount of gas required to execute the transaction, the transaction will consume an amount of gas equal to the gas limit, which is not refunded, before failing and reverting any state changes made by the transaction.
```

File: docs/docs/guides/04_transactions/gas-and-fees.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+const transaction: TransactionForAccessList = {
+	from: '<SENDER ADDRESS>',
+	to: '<RECEIVER ADDRESS>',
+	value: web3.utils.ethUnitMap.ether,
+};
+
+const accessListResult: AccessListResult = await web3.eth.createAccessList(transaction);
+transaction.accessList = accessListResult.accessList;
+transaction.gas = accessListResult.gasUsed;
+```
```
Comment (krzysu) [inline docs/docs/guides/04_transactions/gas-and-fees.md]: I’m not sure about the rest of the team’s opinion, but from my experience, mutating data is generally not a good practice and often leads to unexpected outcomes. The example code would be clearer and more reliable if written as suggested below.
 However, feel free to disregard this suggestion if it doesn’t align with the general style of examples in the documentation.

```suggestion
```ts
const transactionDraft: TransactionForAccessList = {
	from: '<SENDER ADDRESS>',
	to: '<RECEIVER ADDRESS>',
	value: web3.utils.ethUnitMap.ether,
};

const accessListResult: AccessListResult = await web3.eth.createAccessList(transactionDraft);

const transaction: TransactionForAccessList = {
	...transactionDraft,
	accessList: accessListResult.accessList,
	gas: accessListResult.gasUsed,
};
```
```

**Review by krzysu — APPROVED** [review]
Overall, a good read! 👍  I’ve left a few suggestions for improvement.

File: docs/docs/guides/04_transactions/gas-and-fees.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+In addition to the calculated gas cost, an Ethereum transaction can specify an optional priority fee, which is an additional fee that is paid directly to the operator of the node that executes the transaction. Priority fees are intended to incentive node operators to execute transactions. A priority fee is specified as a value in addition to the base fee, which means that the total cost of the priority fee is always a factor of the amount of gas required to execute a transaction.
+
+With the above in mind, the total cost of fees associated with a transaction are calculated as: _units of gas used \* (base fee + priority fee)_.
```
Comment (luu-alex) [inline docs/docs/guides/04_transactions/gas-and-fees.md]:
```suggestion
With the above in mind, the total cost of fees associated with a transaction is calculated as follows:

Total cost of fees = _units of gas used \* (base fee + priority fee)_
```
suggestion - i think writing it out on a new line and as a full math equation can make it easier for the user to read and stands out a bit more

**Review by luu-alex — APPROVED** [review]

**Review by avkos — APPROVED** [review]

**Pull request merged**